packages/devtools_app/lib/src/memory/memory_allocation_table_view.dart

```diff
@@ -92,6 +96,72 @@ class AllocationTableViewState extends State<AllocationTableView>
       final Tracker item = trackerData.selectionNotifier.value.node;
       if (item is TrackerMore) trackerData.expandCallStack(item);
     });
+
+    addAutoDisposeListener(controller.selectTheSearchNotifier, () {
+      if (_trySelectItem()) {
+        setState(() {
+          controller.closeAutoCompleteOverlay();
+        });
+      }
+    });
+
+    addAutoDisposeListener(controller.searchNotifier, () {
+      if (_trySelectItem()) {
+        setState(() {
+          controller.closeAutoCompleteOverlay();
+        });
+      }
+    });
+  }
+
+  /// Search the allocation data for a match (auto-complete).
+  List<String> _allocationMatches(String searchingValue) {
+    final matches = <String>[];
+
+    // Matches that start with searchingValue, most relevant.
+    final startMatches = <String>[];
+
+    for (var allocation in controller.monitorAllocations) {
+      final knownName = allocation.classRef.name;
+      if (knownName.startsWith(searchingValue)) {
+        startMatches.add(knownName);
+      } else if (knownName.contains(searchingValue.toLowerCase())) {
+        matches.add(knownName);
+      }
+    }
+
+    matches.insertAll(0, startMatches);
+    return matches;
+  }
+
+  bool _trySelectItem() {
+    final searchingValue = controller.search;
+    if (searchingValue.isNotEmpty) {
+      if (controller.selectTheSearch) {
+        // Found an exact match.
+        controller.selectItemInAllocationTable(searchingValue);
+        controller.selectTheSearch = false;
+        controller.resetSearch();
```

Comment:
will this affect the search setup on the analysis tab?

Reply:
No, each has their own _trySelectItem.